nifi-nar-bundles/nifi-extension-utils/nifi-hadoop-utils/src/main/java/org/apache/nifi/processors/hadoop/AbstractHadoopProcessor.java

```diff
@@ -321,6 +356,15 @@ public final void abstractOnStopped() {
                 }
             }
 
+            final KerberosUser kerberosUser = resources.getKerberosUser();
+            if (kerberosUser != null) {
+                try {
+                    kerberosUser.logout();
+                } catch (LoginException e) {
+                    getLogger().warn("Error logging out KerberosUser: " + e.getMessage(), e);
```

Comment:
This could be adjusted to use placeholders instead of concatenation. Is there value in including the KerberosUser name in the message?
```suggestion
                    getLogger().warn("Error logging out KerberosUser: {}", e.getMessage(), e);
```